cli/targets/static.js

```diff
@@ -354,8 +354,48 @@ function toJsType(field) {
     return type;
 }
 
+function syntaxForType(type) {
+
+    var syntax = null;
+    var namespace = type;
+
+    while (syntax === null && namespace !== null) {
+        if (namespace.options != null && "syntax" in namespace.options)
+            syntax = namespace.options["syntax"];
+        else
+            namespace = namespace.parent;
+    }
+
+    return (syntax === "proto3") ? "proto3" : "proto2";
+}
+
+function isOptional(field, syntax) {
+
+    // In proto3, optional fields are explicit
+    if (syntax === "proto3")
+        return field.options != null && field.options["proto3_optional"] === true;
+
+    // In proto2, fields are explicitly optional if they are not part of a map, array or oneOf group
+    return field.optional && !(field.partOf || field.repeated || field.map);
```

Comment:
Thinking ahead to editions syntax, should we also gate this `if syntax === proto2`?

Reply:
Done, changes pushed. I'm throwing an error for unknown proto syntax if it is not either "proto2" or "proto3".